src/users/users.service.ts

```diff
@@ -0,0 +1,25 @@
+import { BadRequestException, HttpException, Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from 'src/prisma.service';
+
+@Injectable()
+export class UsersService {
+    constructor(
+        private prisma: PrismaService
+    ){}
+
+    // get a single user
+    async getOne(email): Promise<any>{
+        try{
+                // Check for the user with the id
+            const user = await this.prisma.user.findUnique({where:{
+                email: email
```

Comment:
`getOne` method shouldn't be tied to a named parameter/criteria. Let's pass in an object instead so that different criteria could be used to find one record

```suggestion
    async getOne(criteria: Object): Promise<User>{
        try{
                // Check for the user with the id
            const user = await this.prisma.user.findUnique({ where: 
                { ...criteria, deleted: false
```